std/uni.d

```diff
@@ -7079,10 +7082,31 @@ private int fullCasedCmp(Range)(dchar lhs, dchar rhs, ref Range rtail)
         $(LREF sicmp)
         $(REF cmp, std,algorithm,comparison)
 +/
-int icmp(S1, S2)(S1 str1, S2 str2)
-    if (isForwardRange!S1 && is(Unqual!(ElementType!S1) == dchar)
-    && isForwardRange!S2 && is(Unqual!(ElementType!S2) == dchar))
+int icmp(S1, S2)(S1 r1, S2 r2)
+    if (isForwardRange!S1 && isSomeChar!(ElementEncodingType!S1)
+    && isForwardRange!S2 && isSomeChar!(ElementEncodingType!S2))
 {
+    // auto-decoding special case
+    static if (!is(ElementEncodingType!S1 == dchar))
```

Comment:
I think this is moot, most of std.uni relies on std.utf anyway, let's just put a single byDchar no need to complicate this more.